plugins/modules/system/lvol.py

```diff
@@ -339,11 +346,23 @@ def main():
             size_opt = 'l'
             size_unit = ''
 
+        # LVCREATE(8) -L --size option unit
         if '%' not in size:
-            # LVCREATE(8) -L --size option unit
-            if size[-1].lower() in 'bskmgtpe':
+            if size[-1] in 'BSKMGTPE':
                 size_unit = size[-1].lower()
                 size = size[0:-1]
+                size_divisor = {'B': 1.000, 'S': 1.000, 'K': 1.024, 'M': 1.048, 'G': 1.047, 'T': 1.099, 'P': 1.126, 'E': 1.153}.get(size_unit.upper(), '1.000')
+                if '+' in size:
+                    size_operator = '+'
+                    size = str(float(size[1:]) / float(size_divisor))
+                else:
+                    size = str(float(size) / float(size_divisor))
+            elif size[-1] in 'bskmgtpe':
```

Comment:
@zigaSRC not sure if I understood the difference between upper case units and lower case units - per `lvcreate(8)` it seems that there is none, but the code here seems to treat those cases differently. Could you please clarify? TIA

Reply:
Sorry, this was just a straight copy of the PR from the old repo. I didn't check the code as it was already reviewed there. My mistake, will fix it.